manifests/init.pp

```diff
@@ -47,6 +47,10 @@
       fail("invalid version ${package_version} requested")
     }
 
+    $aio_upgrade_required = ($is_pe == false and $package_version != undef) or
+      ($::aio_agent_version != undef and $package_version != undef and
```

Comment:
Yeah I'm not sure what is happening there and yeah part of the reason I decided to centralize this check is so we can easily update it later if we port the fact over.